mgapi/handler.py

```diff
@@ -0,0 +1,109 @@
+""" Corpus Fetcher
+
+ Loads corpus data from MemeGenerator.net API
+ """
+
+import urllib.request
+import json
+
+
+class MemeGeneratorApiHandler:
+    """ Fetches mgapi data via API """
+    def __init__(self, username, password):
+        """ initializes l/p """
+
+        self._username = username
+        self._password = password
+
+    def _get_request_url(self, action, params=''):
+        """ String builder for API endpoints """
+        url = 'http://{}/{}'.format(
+            'version1.api.memegenerator.net',
+            action
+        )
+
+        for i, param in enumerate(params):
+            url = self._add_url_param(url, param, params[param])
+
+        return url
```

Comment:
[urllib.parse.unsplit()](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunsplit) has also already implemented this.